### ISSUE TITLE
[IMPROVEMENT] Nondestrutive engine upgrade after introduing new methods in engine grpc proxy

### ISSUE THREAD

**Issue by derekbit** [longhorn/longhorn#4833]
## Is your improvement request related to a feature? Please describe

Longhorn starts communicating with engine via grpc proxy instead of legacy CLI from v1.3.0. If new proxy methods are added and upgrade the attached volume's engine, longhorn-manager cannot call the methods to the engine before detaching and re-attaching the volume. It breaks the previous behavior using legacy CLI and introduces a service downtime because of the volume detachment. We need a nondestrutive method for making the engine upgrade trasparent to user workloads.

**Comment by innobead** [comment]
cc @longhorn/dev

**Comment by shuo-wu** [comment]
Do you mean the old longhorn managers are not able to talk with the upgraded new engines?

**Comment by derekbit** [comment]
Yes, if there is any new grpc methods in the engine proxy service.

**Comment by shuo-wu** [comment]
Got it. Right now the gRPC (backward compatibility) can guarantee that the new longhorn manager (client) is able to talk with the old longhorn engine (server) but cannot do it in a reverse way.

**Comment by innobead** [comment]
ref #5150
